### ISSUE TITLE
Autoconf second `make` causes full rebuild

### ISSUE THREAD

**Issue by underwoo** [NOAA-GFDL/FMS#648]
**Describe the bug**
If using the autoconf build, issuing a second `make` command with no changes will cause nearly a full re-build of the FMS library components.  This makes performing a `make install` more cumbersome as a rebuild is not needed.

**To Reproduce**
```
autoreconf -if
configure [options]
make
make # this should do nothing!
```

**Expected behavior**
The second call to `make` should print many lines of 'Nothing to be done for ...'.

**Comment by underwoo** [comment]
Even running a `make check` triggers a rebuild.  This will increase our testing time, and slow our progress.

**Comment by underwoo** [comment]
From looking at the GitHub actions page (https://github.com/NOAA-GFDL/FMS/actions?query=workflow%3A%22Build+libFMS+test%22) I can see a sudden jump about 21 days ago after merging the master branch into the GitHub action branch.  Some back tracking will be needed to see what change is causing this.

**Comment by underwoo** [comment]
Think I got the build/rebuild issue resolved.  The number of changes are more than I originally anticipated, again as I decided to make changes that I have wanted to make for a while.  I hope the final version of the `Makefile.am`s and other autoconf build files are easier to maintain.  

As an added note, I added additional dependencies to have a second `make` build when touching the include files.  While this helps in many cases, a change to an interface in a `*.mod` file will not trigger a rebuild.  Although I think that could be added, but it isn't part of this fix.
